printbuf.c

```diff
@@ -142,14 +177,29 @@ int sprintbuf(struct printbuf *p, const char *msg, ...)
 
 void printbuf_reset(struct printbuf *p)
 {
-  p->buf[0] = '\0';
-  p->bpos = 0;
+  if (p != NULL)
```

Comment:
It's not valid to call this with a NULL.  At most, this could be an assert(), but crashing is perfectly reasonable too.